### PR TITLE
PDI-13367 - ETL Metada Injection does not actualize the Source Step in the Inject Metadata tab when renaming the cited step

### PR DESCRIPTION
**Pull request by yuryilyukevich**
Added the list of listeners changes steps

### CODE REVIEW

File: engine/src/org/pentaho/di/trans/TransMeta.java [hunk 1]
```diff
@@ -6109,4 +6136,45 @@ public void setForcingSeparateLogging( boolean forcingSeparateLogging ) {
...
+      if ( listener.getStepMeta().equals( removeStep ) ) {
+        indexListnerRemove = indexListner;
+      }
+    }
+    if ( indexListnerRemove >= 0 ) {
+      stepChangeListeners.remove( indexListnerRemove );
+    }
+  }
+
+  public void NotifyAllListeners( StepMeta oldMeta, StepMeta newMeta ) {
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/trans/TransMeta.java]: This method should start with a lowercase letter

File: engine/src/org/pentaho/di/trans/TransMeta.java [hunk 2]
```diff
@@ -6109,4 +6136,45 @@ public void setForcingSeparateLogging( boolean forcingSeparateLogging ) {
...
+      }
+    }
+    if ( indexListnerRemove >= 0 ) {
+      stepChangeListeners.add( indexListnerRemove, list );
+    }
+  }
+
+  public void removeStepChangeListener( int p ) {
+    int indexListner = -1;
+    int indexListnerRemove = -1;
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/trans/TransMeta.java]: These variables should say "Listener" not "Listner"

**Comment by mtseu** [comment]
Hey @mattyb149. The request has already been closed - it was sent a bit prematurely.

File: engine/src/org/pentaho/di/trans/TransMeta.java [hunk 3]
```diff
@@ -6109,4 +6136,45 @@ public void setForcingSeparateLogging( boolean forcingSeparateLogging ) {
...
+      if ( listener.getStepMeta().equals( rewriteStep ) ) {
+        indexListnerRemove = indexListner;
+      }
+    }
+    if ( indexListnerRemove >= 0 ) {
+      stepChangeListeners.add( indexListnerRemove, list );
+    }
+  }
+
+  public void removeStepChangeListener( int p ) {
```
Comment (mattyb149) [inline engine/src/org/pentaho/di/trans/TransMeta.java]: Should we return the removed listener?

**Pull request closed without merging**